pvxarray/vtk_source.py

```diff
@@ -189,42 +185,107 @@ def z_index(self, z_index: int):
         self.Modified()
 
     @property
-    def data_range(self):
-        return self.data_array.min(), self.data_array.max()
+    def slicing(self):
+        return self._slicing
 
-    def RequestData(self, request, inInfo, outInfo):
-        # Use open data_array handle to fetch data at
-        # desired Level of Detail
-        try:
-            if self._time is not None:
-                da = self.data_array[{self._time: self.time_index}]
-            else:
-                da = self.data_array
+    @slicing.setter
+    def slicing(self, slicing: int):
+        self._slicing = slicing
+        self.Modified()
 
-            if self._z and self._z_index is not None:
-                da = da[{self._z: self.z_index}]
+    @property
+    def sliced_data_array(self):
+        if self._sliced_data_array is None:
+            self._compute_sliced_data_array()
+        return self._sliced_data_array
+
+    @property
+    def persisted_data(self):
+        if self._persisted_data is None:
+            self._persisted_data = self.sliced_data_array.persist()
+        return self._persisted_data
+
+    @property
+    def mesh(self):
+        if self._mesh is None:
+            self._compute_mesh()
+        return self._mesh
 
+    @property
+    def data_range(self):
+        da = self.persisted_data
+        return da.min(), da.max()
+
+    def resolution_to_sampling_rate(self, data_array):
+        """Convert percentage to sampling rate."""
+        shape = np.array(data_array.shape)
+        n = np.floor(shape * self._resolution)
+        rate = np.ceil(shape / n).astype(int)
+        return np.pad(rate, (0, 3 - len(rate)), mode="constant")
+
+    def _compute_sliced_data_array(self):
+        if self.data_array is None:
+            self._sliced_data_array = None
+            return None
+
+        if self._time is not None:
+            da = self.data_array[{self._time: self.time_index}]
+        else:
+            da = self.data_array
+
+        if self._z and self._z_index is not None:
+            da = da[{self._z: self.z_index}]
+
+        if self._slicing:
+            indexing = {}
+            for axis in [
+                self.x,
+                self.y,
+                self.z,
+            ]:
+                if axis in self.slicing:
```

Comment:
Very minor, we are using both `self._slicing` and `self.slicing` in the span of a few lines, we should settle on either one for consistency

Reply:
Thanks, I missed the underscore there. Changed in [e25e281](https://github.com/pyvista/pyvista-xarray/pull/58/commits/e25e281f257ccff85b58b7070d95e2258139d17f)